wagtail/wagtailcore/templatetags/wagtailcore_tags.py

```diff
@@ -35,9 +35,12 @@ def wagtail_version():
 
 @register.filter
 def richtext(value):
-    if value is not None:
-        html = expand_db_html(value)
-    else:
+    if isinstance(value, RichText):
+        # passing a RichText value through the |richtext filter should have no effect
+        return value
+    elif value is None:
         html = ''
+    else:
+        html = expand_db_html(value)
 
     return mark_safe('<div class="rich-text">' + html + '</div>')
```

Comment:
I considered that (especially as it would eliminate the special case for None...) but didn't want to risk making any breaking changes, since refactoring the richtext filter wasn't the primary aim of this PR. (Changing the filter to return a RichText object would have been quite elegant, but you never know if someone's done something like `foo|richtext|upper` which would break. So if I cast the result to a string, do I want `force_text` or `str`... and so on :-) )